app/cli/internal/telemetry/posthog/posthog.go

```diff
@@ -53,14 +53,20 @@ type Tracker struct {
 
 // TrackEvent sends an event to the PostHog server.
 func (p *Tracker) TrackEvent(_ context.Context, eventName string, id string, tags telemetry.Tags) error {
+	defer p.client.Close()
```

Comment:
why is this needed? To send the event once the function finishes?

Reply:
Yes and to ensure the client is properly closed.